sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -493,11 +495,12 @@ class Dataset[T] private[sql](
    *
    * @since 1.6.0
    */
-  def joinWith[U](other: Dataset[U], condition: Column): Dataset[(T, U)] = {
+  def joinWith[U](other: Dataset[U], condition: Column, joinType: String): Dataset[(T, U)] = {
```

Comment:
you need to update the documentation to explain what options are available for joinType.